pkg/exec/devfile.go

```diff
@@ -12,160 +13,62 @@ import (
 )
 
 // ExecuteDevfileCommandSynchronously executes the devfile init, build and test command actions synchronously
-func ExecuteDevfileCommandSynchronously(client ExecClient, exec common.Exec, commandName string, compInfo adaptersCommon.ComponentInfo, show bool, machineEventLogger machineoutput.MachineEventLoggingClient) error {
-	var s *log.Status
-
+func ExecuteDevfileCommandSynchronously(client ExecClient, exec common.Exec, compInfo adaptersCommon.ComponentInfo, show bool, machineEventLogger machineoutput.MachineEventLoggingClient) error {
```

Comment:
it would be great if we can get the argument count to less then or equal to 4?

Reply:
maybe consider options pattern? https://www.sohamkamani.com/golang/options-pattern

Reply:
I considered bundling all the parameters into a struct. Will see if I can come up with something that makes sense.

Reply:
The problem that I see with relying on functional options is that it makes for a not very approachable API: if you don't have an example of how to use it, you'd be hard-pressed to figure out how to use it just by reading the function's signature. That pattern doesn't seem pervasive either as I don't recall having encountered it in my, granted limited, go development experience.

Reply:
I have significantly simplified the API by making these functions methods of the adapter via a generic adapter.